packages/@vuepress/core/lib/client/root-mixins/updateMeta.js

```diff
@@ -35,10 +35,26 @@ export default {
 
     getMergedMetaTags () {
       const pageMeta = this.$page.frontmatter.meta || []
+      const templateMeta = []
+      if (typeof document !== 'undefined') {
+        const templateMetaTags = document.querySelectorAll('meta')
+        templateMetaTags.forEach(tag => {
+          const names = tag.getAttributeNames()
+          if (names.length === 1) {
+            const attr = tag.getAttributeNames()[0]
+            const obj = {}
+            obj[attr] = tag.getAttribute(attr)
+            templateMeta.push(obj)
+          } else if (names.every(e => ['name', 'content'].includes(e))) {
```

Comment:
wondering about the why these if statements are necessary, simply assign each attribute with their name should be fine:
e.g.
```
        const obj = {}
        for (const key of e.getAttributeNames()) {
          obj[key] = e.getAttribute(key)
        }
        return obj
```

Reply:
> wondering about why these if statements are necessary, simply assign each attribute with their name should be fine?

e.g. https://github.com/vuejs/vuepress/pull/2360/files#diff-436014f2c7425af1edd74dfd675d069eR42

Reply:
@NeverBehave some meta tags doesn't have name attribute. then, i wrote these if statements.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/meta

Reply:
I am a bit confused, I think the `getAttributeNames()` will return all names from the elements: 
```
> a
<meta charset="utf-8">

> a.getAttributeNames()
Array [ "charset" ]
```

So basically loop through each of these attributes should be fine? What exactly the name you reference to? If `<meta name=""/>` then it is also included in the function returns. 

```
> document.querySelectorAll('meta')[1]
<meta http-equiv="X-UA-Compatible" content="IE=Edge">

> document.querySelectorAll('meta')[1].getAttributeNames()
Array [ "http-equiv", "content" ]
```
The example I provided previous did pass the test tho, correct me if I miss anything

Reply:
sorry, i misunderstood. thank you for the information.